### DISCUSSION TITLE
Assigning Contacts to DNS records..

### DISCUSSION THREAD

**Discussion by ThomasADavis** [sys4/netbox-plugin-dns#369] [Ideas]
At my place of work, they have historically assigned a set of contacts to DNS records.  This allows them to directly contact the requester(s)/tenants if something goes bad.

In netbox, contacts can be assigned to objects.  For instance, ipam/ip-address ojbects can have a contact assigned to it, even though you can not do this on the IPAM web interfaces.  You use the API to do this.  This relationship does however show up on the IPAM/IPaddress page.

To assign contacts, you need an object_type, and an object_id.  The id I can easily get - it's what returned from api calls.  The object_type, for a dns record, I'm still looking for..  if it exists.

If it does exists, what is it? 

If it doesn't exist, can a object_type for this type of operation be registered?

I currently assign contacts to IP addresses, but I'd like to be able to assign contacts to DNS records of all types.

**Comment by peteeckel** [comment]
Hi @ThomasADavis, 

that's interesting (as always :-)). I never tried it, but does this help you:

```
>>> from netbox_dns.models import Record
>>> from core.models import ObjectType
>>> ObjectType.objects.get_for_model(Record)
<ObjectType: NetBox DNS | record>
```

There's also `ContentType` in Django:
```
>>> from django.contrib.contenttypes.models import ContentType
>>> ContentType.objects.get_for_model(Record)
<ContentType: NetBox DNS | record>
```

They look pretty similar but probably aren't - You could try either of them and see if that gets you going.

**Comment by peteeckel** [comment]
OK, just to preserve the conversation from Slack: Here's how to enable contact assignments for plugin models.

Depending on how you create your URLs, there are two or three steps:

1. Have the plugin models you want to enable contact assignments for use `ContactsMixin` in the `models` module:
```python
from netbox.models.features import ContactsMixin
...
class MyModel(ContactsMixin, NetBoxModel):
...
```

2. Create a view for the contacts in the `views` module, have it inherit from `ObjectContactsView` and register the it for the model using `register_model_view` :
```python
from utilities.views import register_model_view
from tenancy.views import ObjectContactsView
...
@register_model_view(MyModel, "contacts")
class MyModelContactsView(ObjectContactsView):
    queryset = MyModel.objects.all()
...
```

3. If you are using `get_model_urls()` in your URL definition you're already done. If not, you need to manually register an URL in the `urls` module:
```python
from mymodel.views import MyModelContactsView
...
    path("mymodels/<int:pk>/contacts/", MyModelContactsView.as_view(), name="mymodel_contacts"),
...
```

**Comment by ghost** [comment]
would these steps allow assigning contacts to dns records within the gui?

**Reply by peteeckel** [reply]
> would these steps allow assigning contacts to dns records within the gui?

<img width="1224" alt="Screenshot 2024-09-02 at 18 30 35" src="https://github.com/user-attachments/assets/0998367b-d7b6-4fad-9ba9-312f62cca390">

:-)

**Reply by peteeckel** [reply]
So, yes, it does. It's exactly the same as for the stock models.
